### PR TITLE
feat: add update & delete trigger

### PR DESCRIPTION
**Pull request by smomin**
- add update and delete triggers
- update extension params

### CODE REVIEW

**Comment by Salakar** [comment]
cc @dackers86 for additional review

**Comment by smomin** [comment]
@Salakar I am waiting for him to accept the invite to this repo before i can ask for a review.

**Comment by i14h** [comment]
@smomin first feedback is to include the compiled js code in the commit,

File: extension.yaml [hunk 1]
```diff
@@ -31,46 +31,54 @@ billingRequired: false
...
       Specify the Algolia Application Id.
     type: string
     required: true
     immutable: true   
 
-  - param: API_KEY
+  - param: ALGOLIA_API_KEY
     label: Algolia API Key
     description: >-
       Specify the Algolia API Key that has modification permission to the index.
```
Comment (dackers86) [inline extension.yaml:84]: Could we perhaps specify which api key, this left me a bit confused on installation as my dashboard  options are...

Search-only
Admin
Usage
Monitoring

I guessed search only.

**Comment by dackers86** [comment]
Additionally, would it be possible to get a quick update on how to run the extension.

For example, what data would typically be entered into the `search` collection to start a search?

File: extension.yaml (hunk 1 above)
Comment (i14h) [reply extension.yaml:84]: I agree with the need for clarification - it should be admin api key or another one with  write access

File: functions/src/extract.ts [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+ * limitations under the License.
+ */
+
+import { DocumentSnapshot } from 'firebase-functions/lib/providers/firestore';
+
+import config from './config';
+import * as logs from './logs';
+
+const trim = /^[\s\uFEFF\xA0]+|[\s\uFEFF\xA0]+$/g;
+console.log(`fields are defined: ${config?.fields}`);
```
Comment (Salakar) [inline functions/src/extract.ts:23]: Is this log intended?

File: functions/src/logs.ts [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+
+import config from "./config";
+
+const obfuscatedConfig = {
+  ...config,
+  algoliaAPIKey: "********",
+};
+
+export const init = () => {
+  console.info("Initializing mod with configuration", obfuscatedConfig);
```
Comment (Salakar) [inline functions/src/logs.ts:25]:
```suggestion
  console.info("Initializing extension with configuration", obfuscatedConfig);
```

File: functions/src/logs.ts [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+};
+
+export const start = () => {
+  console.info("Started mod execution with configuration", obfuscatedConfig);
```
Comment (Salakar) [inline functions/src/logs.ts:29]:
```suggestion
  console.info("Started extension execution with configuration", obfuscatedConfig);
```

File: functions/src/logs.ts [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+export const updateIndex = (id: string, data: object) => {
+  console.info(`Updating existing Algolia index for document ${id}`, data);
+};
+
+export const deleteIndex = (id: string) => {
+  console.info(`Deleting existing Algolia index for document ${id}`);
+};
+
+export const fieldNotExist = (field: string) => {
+  console.warn(`The field "${field}" was specified in the mod config but was not found on collection data.`);
```
Comment (Salakar) [inline functions/src/logs.ts:49]:
```suggestion
  console.warn(`The field "${field}" was specified in the extension config but was not found on collection data.`);
```

File: extension.yaml (hunk 1 above)
Comment (smomin) [reply extension.yaml:84]: Thanks.  I will update the description to be more specific.

File: functions/src/extract.ts (hunk 2 above)
Comment (smomin) [reply functions/src/extract.ts:23]: no.  i will take it out.

**Comment by smomin** [comment]
I am going to move forward and merge this PR.

**Pull request merged**